### PR TITLE
CATL-1593: Show inactive roles

### PR DESCRIPTION
**Pull request by reneolivo**
## Overview
This PR allows displaying past case roles that have been removed from the case.

## Before
pr-before
Past roles are not visible and there's no way to make them visible.

## After
gif

## Technical Details

* The Case Details People tab directive was refactored in order to follow coding standards.
* A new service was added: People's Tab Role service. This service has the logic for all of the following features, taken from the main directive:
  * Controls the list of roles and clients that are going to be displayed on the people's tab.
  * Displays unassigned case roles.
  * Supports role filtering by letter and role types.
  * Paginates the list of roles when it's longer than 25 records.

This new service was implemented on the People's Tab directive and the old code that was ported was removed.

Some more notes:

* A new `Display previous case role assignments` checkbox element was added to the people's tab template. The `_case-peoples.scss` file was updated to add styles for the checkbox.
* The `center-block` class was removed from the first paging component because it breaks due to the newly inserted checkbox element.
* A `Past` badge element is appended to roles when they had been removed.
* The `getSelectedContacts` function was refactored to accommodate to the new service's structure.
* The `get-case-query-params.factory.js` service was updated so it returns all case relationships even if they are not active. The relationship's contact is also included in case the relationship is no longer active since the contact would not be available in the case details. 
* The `getUniqueCaseRelationships` function of the roles service is needed because multiple relationships for different clients might be returned, but we only need a single result for the role contact.

## Bug Fixes

* CATL-1807: Fix Add Case Role button https://github.com/compucorp/uk.co.compucorp.civicase/pull/588
* CATL-1852: Fix limit for adding roles https://github.com/compucorp/uk.co.compucorp.civicase/pull/601

### CODE REVIEW

**Comment by deb1990** [comment]
@reneolivo Please add labels.

**Comment by deb1990** [comment]
@reneolivo Is the UI discussed with @fran-compucorp ? Personally I would prefer something like "Inactive" instead of "Past". But if it is discussed its fine.
Also should not we show all the Inactive roles in the end?

File: ang/civicase/case/details/people-tab/directives/case-details-people-tab.directive.html [hunk 1]
```diff
@@ -74,14 +74,45 @@ <h3>{{ ts('%1 Contacts', {1: getCountOfRolesWithContacts(roles)}) }}</h3>
...
+          ng-class="{active: letter === rolesAlphaFilter}"
+          ng-click="setLetterFilter(letter, 'roles')">
+          {{ letter }}
+        </a>
         <a href class="all" ng-click="setLetterFilter('', 'roles')">All</a>
       </div>
+      <div class="civicase__people-tab__inactive-filter">
+        <span
+          class="civicase__checkbox" role="checkbox" tabindex="0"
+          aria-checked="{{}}" aria-labelledby="Deleted Cases"
```
Comment (deb1990) [inline ang/civicase/case/details/people-tab/directives/case-details-people-tab.directive.html:89]: The value of aria-checked looks weird. Is this supposed to be `aria-checked="{{}}"`?

File: ang/civicase/case/details/people-tab/directives/case-details-people-tab.directive.html [hunk 2]
```diff
@@ -26,11 +26,11 @@
           </li>
```
Comment (deb1990) [inline ang/civicase/case/details/people-tab/directives/case-details-people-tab.directive.html:26]: Thanks @reneolivo . But can we refactor this a little bit more, by breaking the HTML into smaller chunks. No need to create directives for now, but will be great if we can use ng-include and divide the file into Header, Footer, Filter etc.

File: ang/civicase/case/details/people-tab/directives/case-details-people-tab.directive.js [hunk 3]
```diff
@@ -44,22 +44,22 @@
...
    * @param {object} ts ts
    * @param {object} RelationshipType RelationshipType
-   * @param {Function} isTruthy service to check if value is truthy
    * @param {boolean} civicaseSingleCaseRolePerType if a single case role can be assigned per type
    * @param {object} dialogService A reference to the dialog service
    * @param {Function} removeDatePickerHrefs Removes date picker href attributes
    */
   function civicaseViewPeopleController ($q, $scope, allowMultipleCaseClients,
-    civicaseCrmApi, DateHelper, ts, RelationshipType, isTruthy, civicaseSingleCaseRolePerType,
+    civicaseCrmApi, civicasePeopleTabRoles, DateHelper, ts, RelationshipType, civicaseSingleCaseRolePerType,
```
Comment (deb1990) [inline ang/civicase/case/details/people-tab/directives/case-details-people-tab.directive.js:62]: This is still more than 80 chars. Please break the line.

File: ang/civicase/case/details/people-tab/directives/case-details-people-tab.directive.js [hunk 4]
```diff
@@ -44,22 +44,22 @@
...
     dialogService, removeDatePickerHrefs) {
     // The ts() and hs() functions help load strings for this module.
     var CONTACT_CANT_HAVE_ROLE_MESSAGE = ts('Case clients cannot be selected for a case role. Please select another contact.');
```
Comment (deb1990) [inline ang/civicase/case/details/people-tab/directives/case-details-people-tab.directive.js:65]: Can we move the message strings into a peoples-tab-constants file?

File: scss/components/_case-peoples.scss [hunk 5]
```diff
@@ -188,6 +188,14 @@
   }
 }
 
+.civicase__people-tab__inactive-filter {
+  .civicase__checkbox {
+    display: inline-block;
+    margin-right: 5px;
```
Comment (deb1990) [inline scss/components/_case-peoples.scss:194]: Lets use `$basic-gutter`.

File: scss/components/_case-peoples.scss [hunk 6]
```diff
@@ -188,6 +188,14 @@
...
+    top: 4px;
```
Comment (deb1990) [inline scss/components/_case-peoples.scss:195]: Cant we use `5px` or `$basic-gutter`?

**Review by deb1990 — CHANGES_REQUESTED** [review]

**Comment by deb1990** [comment]
Also for the UI, the checkbox should be right aligned.

**Comment by reneolivo** [comment]
@deb1990 :

>  Personally I would prefer something like "Inactive" instead of "Past". But if it is discussed its fine.

"Past" is part of the specs. It said that an icon can be used, but I'm not sure which one to pick.

> Also should not we show all the Inactive roles in the end?

What do you mean?

> Also for the UI, the checkbox should be right aligned.

Manage-Cases-c8008 (3)
done!

**Review by deb1990 — APPROVED** [review]

**Pull request merged**